### PR TITLE
Refactor `add_links` in JSONAPI adapter.

### PR DESCRIPTION
**Pull request by beauby**
+ Moved the `paginated?` method in the `ArraySerializer`
+ Refactored `add_links` into `links_for` in order to be consistent with the refactor of the adapter (#1103)

### CODE REVIEW

File: lib/active_model/serializer/array_serializer.rb [hunk 1]
```diff
@@ -29,6 +29,12 @@ def json_key
         key = root || @serializers.first.try(:json_key) || object.try(:name).try(:underscore)
         key.try(:pluralize)
       end
+
+      def paginated?
```
Comment (bacarini) [inline lib/active_model/serializer/array_serializer.rb]: :+1:

File: lib/active_model/serializer/adapter/json_api.rb [hunk 2]
```diff
@@ -19,35 +17,50 @@ def initialize(serializer, options = {})
...
+            result = self.class.new(s, @options.merge(fieldset: @fieldset)).serializable_hash(options)
+            hash[:data] << result[:data]
+
+            if result[:included]
+              hash[:included] ||= []
+              hash[:included] |= result[:included]
+            end
+          end
+
+          if serializer.paginated?
```
Comment (bacarini) [inline lib/active_model/serializer/adapter/json_api.rb]: this check could be inside `links_for` method. Today we have just pagination link, but for the future other links need be included as well.

something like that
```
hash[:links]  = links_for(serializer, options)
```

```
def links_for(serializer, options)
  return {} unless serializer.paginated?
  JsonApi::PaginationLinks.new(serializer.object, options[:context]).serializable_hash(options)
end
```

Is that make sense?

Comment (beauby) [inline lib/active_model/serializer/adapter/json_api.rb]: The thing is, as with `relationships` and `include`, we do not want to add the corresponding json key if not requested.

Comment (bacarini) [inline lib/active_model/serializer/adapter/json_api.rb]: :+1:

**Comment by joaomdmoura** [comment]
Nice, I'm merging this one :smile:

**Pull request merged**